src/stormpath.social-login.js

```diff
@@ -0,0 +1,197 @@
+
+  /**
+   * @ngdoc object
+   *
+   * @name stormpath.socialLoginService.$socialLoginProvider
+   *
+   * @description
+   *
+   * Provides the {@link stormpath.socialLoginService.$socialLogin $socialLogin} service.
+   *
+   * Currently, this provider does not have any configuration methods.
+   */
+  .config(['$injector', 'STORMPATH_CONFIG', function $socialLoginProvider($injector, STORMPATH_CONFIG) {
+    /**
+     * @ngdoc object
+     *
+     * @name stormpath.socialLoginService.$socialLogin
+     *
+     * @description
+     *
+     * The social login service provides methods for letting users logging in with Facebook, Google, etc.
+     */
+    var socialLoginFactory = ['$http', '$q', '$injector', function socialLoginFactory($http, $q, $injector) {
+      return new SocialLoginService(STORMPATH_CONFIG, $injector, $http, $q);
+    }];
+
+    $injector.get('$provide').factory(STORMPATH_CONFIG.SOCIAL_LOGIN_SERVICE_NAME, socialLoginFactory);
+  }])
+
+  /**
+   * @ngdoc object
+   *
+   * @name stormpath.socialLoginService.$spJsLoader
+   *
+   * @description
+   *
+   * The `$spJsLoader` provides a method for loading scripts during runtime.
+   * Used by the social provider services to load their SDKs.
+   */
+  .factory('$spJsLoader', ['$q', function($q) {
+    return {
+      load: function load(id, src, innerHTML) {
+        var deferred = $q.defer();
+        var firstJsElement = document.getElementsByTagName('script')[0];
+        var jsElement = document.createElement('script');
+
+        if (document.getElementById(id)) {
+          deferred.resolve();
+        } else {
+          jsElement.id = id;
+          jsElement.src = src;
+          jsElement.innerHTML = innerHTML;
+          jsElement.onload = deferred.resolve;
+
+          firstJsElement.parentNode.insertBefore(jsElement, firstJsElement);
+        }
+
+        return deferred.promise;
+      }
+    };
+  }])
+
+  /**
+   * @ngdoc directive
+   *
+   * @name stormpath.spSocialLogin:spSocialLogin
+   *
+   * @description
+   *
+   * Add this directive to a button or link in order to authenticate using an OAuth provider. The value should be an OAuth provider id such as google or facebook.
+   *
+   * @example
+   *
+   * <pre>
+   * <div class="container">
+   *   <button sp-social-login="facebook" sp-scope="public_profile,email">Login with Facebook</button>
+   * </div>
+   * </pre>
+   */
+  .directive('spSocialLogin', ['$socialLogin', '$auth', function($socialLogin, $auth) {
+    return {
+      link: function(scope, element, attrs) {
+        var providerService;
+        var parentScope = scope.$parent;
+
+        $socialLogin.getProviders().then(function(providers) {
+          var provider = providers[attrs.spSocialLogin];
+
+          if (provider && provider.service) {
+            providerService = provider.service;
+            providerService.init(element);
+          }
+        });
+
+        element.click(function() {
```

Comment:
`element.click` appears to require the proper jQuery library, I think if we use `element.bind('click')` we can avoid that (using angular's jqLite)